src/core/gltf2exporter/dracocompressor_p.cpp

```diff
@@ -0,0 +1,314 @@
+/*
+    dracocompressor_p.cpp
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Juan Jose Casafranca <juan.casafranca@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "dracocompressor_p.h"
+#include "gltf2exporter_p.h"
+#include <Qt3DRender/QGeometry>
+#include <Qt3DRender/QAttribute>
+#include <Qt3DRender/QBuffer>
+#include <draco/mesh/mesh.h>
+#include <draco/compression/encode.h>
+#include <draco/attributes/geometry_attribute.h>
+#include <draco/attributes/geometry_indices.h>
+#include <draco/io/mesh_io.h>
+#include <QVector>
+#include <QFile>
+#include <QLoggingCategory>
+#include <QDataStream>
+#include <fstream>
+#include <iterator>
+#include <iostream>
+
+QT_BEGIN_NAMESPACE
+Q_LOGGING_CATEGORY(dracocompressor, "DracoCompressor")
+namespace Kuesa::DracoCompression {
```

Comment:
clang complains nested namespaces are a C++17 extension. We're targeting c++11

Reply:
done